### PR TITLE
LZ-21

### PR DESCRIPTION
**Pull request by matwodecki**
Eksportowanie do pliku JSON

### CODE REVIEW

**Comment by mimagiera** [comment]
dodaj export csv

File: front/src/components/surveyExport.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import axios from 'axios'
+import React, {Component} from "react";
+import {Button} from "reactstrap";
+import {Link} from "react-router-dom";
+
+class SurveyExport extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            survey: -1,
```
Comment (agsta) [inline front/src/components/surveyExport.js]: This integer value might be confusing - maybe an empty string `""` would be better?

File: front/src/components/surveyExport.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            frontUrl: "http://localhost:3000",
+            serverUrl: "http://localhost:8080"
+        }
+    }
+
+    componentDidMount() {
+        let survey_id = window.location.href
+        survey_id = survey_id.replace(this.state.frontUrl + "/surveyexport/", "")
```
Comment (agsta) [inline front/src/components/surveyExport.js]: - In the first line the variable is called `survey_id,` but actually the whole url is assigned.
- You can use `this.props.match.params.id` to extract id.

File: front/src/components/surveyExport.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        axios.get(this.state.serverUrl + "/surveys/" + survey_id)
+            .then((response) => {
+                this.setState({survey: response.data})
+            })
```
Comment (agsta) [inline front/src/components/surveyExport.js]: - Please add `{headers: {authorization: AuthenticationService.getAuthToken()}}` to all your axios methods - right now all methods return error 401. 
- Catching errors is missing (in all axios methods)

File: front/src/components/surveyExport.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                <h1>Survey {this.getSurveyName()}</h1>
+                <Button onClick={this.downloadJSON}>Download in JSON</Button>
+                <Button onClick={this.downloadCSV}>Download in CSV</Button>
+            </div>
+        )
+    }
+
+    getSurveyName() {
+        return this.state.survey.name
+    }
```
Comment (agsta) [inline front/src/components/surveyExport.js]: This function is only used in one place so maybe it's not necessary (?)

File: front/src/components/surveyExport.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                let blob = new Blob([JSONResult])
+                let url = window.URL.createObjectURL(blob);
+                let a = document.createElement('a');
+                a.href = url;
+                a.download = 'export.json';
+                a.click();
+            })
+    }
+
+    downloadCSV() {
```
Comment (agsta) [inline front/src/components/surveyExport.js]: This method is too long - try to divide it into shorter ones.

File: front/src/components/surveys.js [hunk 6]
```diff
@@ -28,11 +29,20 @@ class Surveys extends Component {
     displaySurveys = (surveys) => {
 
         if (!surveys.length) return;
+        console.log(window.location.href)
+        console.log(this.state.frontUrl + "/surveys")
```
Comment (agsta) [inline front/src/components/surveys.js]: Please delete unnecessary console.log functions.

**Review by agsta — CHANGES_REQUESTED** [review]
- Please add headers with authorization field to all your axios methods - currently all of them return 401 error
- It would be better to add the 'Download' buttons in the survey list view (/surveyexport) instead of redirecting to another page with just two buttons

File: front/src/components/surveyExport.js (hunk 5 above)
Comment (Mazarizev) [reply front/src/components/surveyExport.js:62]: Mniejsze metody nie są potrzebne. Wystarczy pozbyć się kilku zmiennych, które są tylko raz wykorzystywane.

File: front/src/components/surveys.js [hunk 7]
```diff
@@ -28,11 +29,20 @@ class Surveys extends Component {
     displaySurveys = (surveys) => {
 
         if (!surveys.length) return;
```
Comment (ghost) [inline front/src/components/surveys.js]: Może jakiś komunikat o braku ankiet? Wyświetlenie `<p>No surveys</p>` wiele wyjaśni.

File: front/src/components/surveyExport.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                this.setState({survey: response.data})
+            })
+    }
+
+    render() {
+        return (
+            <div className="container">
+                <br/>
+                <br/>
+                <br/> {/*TODO: To nie powinno być w ten sposób, ale póki co działa - poprawię później*/}
```
Comment (ghost) [inline front/src/components/surveyExport.js]: Niech TODO informuje **co** planujesz zrobić, a nie **że** planujesz.

**Review by ghost — APPROVED** [review]

**Review by agsta — APPROVED** [review]

**Pull request merged**